v3/pkg/application/application_darwin.go

```diff
@@ -377,3 +384,42 @@
 		ctx: eventContext,
 	}
 }
+
+func (m *macosApp) setStartAtLogin(enabled bool) error {
+	exe, err := os.Executable()
+	if err != nil {
+		return errors.Wrap(err, "Error running os.Executable:")
+	}
+
+	binName := filepath.Base(exe)
+	if !strings.HasSuffix(exe, "/Contents/MacOS/"+binName) {
+		return fmt.Errorf("app needs to be running as package.app file to start at login")
+	}
+
+	appPath := strings.TrimSuffix(exe, "/Contents/MacOS/"+binName)
+	var command string
+	if enabled {
+		command = fmt.Sprintf("tell application \"System Events\" to make login item at end with properties {name: \"%s\",path:\"%s\", hidden:false}", binName, appPath)
+	} else {
+		command = fmt.Sprintf("tell application \"System Events\" to delete login item \"%s\"", binName)
+	}
+
+	cmd := exec.Command("osascript", "-e", command)
+	_, err = cmd.CombinedOutput()
+	return err
+}
+
+func (m *macosApp) canStartAtLogin() bool {
+	bundleID := mac.GetBundleID()
+	if bundleID == "" {
+		return false
+	}
+
+	exe, err := os.Executable()
+	if err != nil {
+		return false
+	}
+
+	binName := filepath.Base(exe)
+	return strings.HasSuffix(exe, "/Contents/MacOS/"+binName)
+}
```

Comment:
@leaanthony Most likely it's not needed. I had tested it on my machine. I'll do one thing to be double sure, I'll build and sign it using GitHub Actions and then will check how it behaves on my machine.

Reply:
It may be needed on later versions of macOS. The rabbit may be on to something..... 
https://developer.apple.com/documentation/bundleresources/information-property-list/nsappleeventsusagedescription

Reply:
@leaanthony I build the package on github action and tested on my machine, it seems to be working without `NSAppleEventsUsageDescription`. Do you still want me to add it ? 

```
- Macbook Air M1 running on 15.1.1 (24B91)
 ```

`Relevant Screenshot`

<img width="580" alt="Screenshot 2024-11-28 at 9 41 01 PM" src="https://github.com/user-attachments/assets/2b37bfeb-9b2e-4913-bcda-2b04ca6b68d9">
<img width="506" alt="Screenshot 2024-11-28 at 9 41 07 PM" src="https://github.com/user-attachments/assets/d2235453-1b6a-4256-8da6-f711df20da0b">

Reply:
If it works, then that's all good